pkg/scheduler/framework/plugins/dynamicresources/extendeddynamicresources.go

```diff
@@ -391,3 +403,164 @@ func createRequestMappings(claim *resourceapi.ResourceClaim, pod *v1.Pod) []v1.C
+				return fmt.Errorf("get resourceclaim %s/%s: %w", claim.Namespace, claim.Name, err)
+			}
+			claim = updatedClaim
+		} else {
+			refreshClaim = true
+		}
+		// Remove the finalizer to unblock removal first.
+		builtinControllerFinalizer := slices.Index(claim.Finalizers, resourceapi.Finalizer)
+		if builtinControllerFinalizer >= 0 {
+			claim.Finalizers = slices.Delete(claim.Finalizers, builtinControllerFinalizer, builtinControllerFinalizer+1)
+		}
+
+		_, err := pl.clientset.ResourceV1().ResourceClaims(claim.Namespace).Update(ctx, claim, metav1.UpdateOptions{})
+		if err != nil {
+			return fmt.Errorf("update resourceclaim %s/%s: %w", claim.Namespace, claim.Name, err)
+		}
+		return nil
+	})
+	if retryErr != nil {
+		return retryErr
+	}
+
+	logger.V(5).Info("Delete", "resourceclaim", klog.KObj(claim))
+	err := pl.clientset.ResourceV1().ResourceClaims(claim.Namespace).Delete(ctx, claim.Name, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// createExtendedResourceClaimInAPI creates an extended resource claim in the API server.
+func (pl *DynamicResources) createExtendedResourceClaimInAPI(
+	ctx context.Context,
+	logger klog.Logger,
+	pod *v1.Pod,
+	nodeName string,
+	state *stateData,
+) (*resourceapi.ResourceClaim, error) {
+	logger.V(5).Info("preparing to create claim for extended resources", "pod", klog.KObj(pod), "node", nodeName)
+
+	// Get the node-specific claim that was prepared during Filter phase
+	nodeAllocation, ok := state.nodeAllocations[nodeName]
+	if !ok || nodeAllocation.extendedResourceClaim == nil {
+		return nil, fmt.Errorf("extended resource claim not found for node %s", nodeName)
+	}
+
+	claim := nodeAllocation.extendedResourceClaim.DeepCopy()
+	logger.V(5).Info("create claim for extended resources", "pod", klog.KObj(pod), "node", nodeName, "resourceclaim", klog.Format(claim))
+
+	claim.Status.Allocation = nil
+	claim.Name = ""
+	claim.UID = ""
+
+	createdClaim, err := pl.clientset.ResourceV1().ResourceClaims(claim.Namespace).Create(ctx, claim, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("create claim for extended resources %v: %w", klog.KObj(claim), err)
+	}
+
+	logger.V(5).Info("created claim for extended resources", "pod", klog.KObj(pod), "node", nodeName, "resourceclaim", klog.Format(createdClaim))
+
+	if err := state.claims.updateExtendedResourceClaim(createdClaim); err != nil {
+		return nil, fmt.Errorf("internal error: update extended ResourceClaim: %w", err)
+	}
+
+	return createdClaim, nil
+}
+
+// waitForExtendedClaimInAssumeCache polls the assume cache until the extended resource claim
+// becomes visible. This is necessary because extended resource claims are created in the API
+// server, and the informer update may not have reached the assume cache yet.
+//
+// AssumeClaimAfterAPICall returns ErrNotFound when the informer update hasn't arrived,
+// so we poll with a timeout.
+func (pl *DynamicResources) waitForExtendedClaimInAssumeCache(
```

Comment:
i think this function is new, but it does not show diff, it shows it is purely added.